app/submitter/converter.py

```diff
@@ -82,14 +83,30 @@ def convert_answers_to_data(answer_store, questionnaire_schema):
     for answer in answer_store.answers:
         item = questionnaire_schema.get_item_by_id(answer['answer_id'])
         value = answer['value']
+
         if item is not None and value is not None:
-            if item.code not in data:
-                data[item.code] = _encode_value(value)
+            if not isinstance(item, CheckboxAnswer) or any('q_code' not in option for option in item.options):
+
+                if item.code not in data:
```

Comment:
Agree for readability, however I don't think I can get rid of the fact I would be passing the array, modifying it and passing it back, which I want to avoid. The ordered dict (data) has to sit outside the for loop for **answer_store.answers**, and the problem is there are checks in the later code to make sure the data doesn't already exist **item.code not in data**, and if it does exist what type it is **isinstance(data[item.code], list)**